tests/hla/test_extract.py

```diff
@@ -0,0 +1,179 @@
+            reference_table = Table.read(reference_catalog, format=tab_format)
+            num_expected = len(reference_table)
+
+            # Perform matching
+            match = tweakwcs.TPMatch(searchrad=5, separation=0.1, tolerance=5, use2dhist=True)
+            ridx, iidx = match(reference_table, imcat, wcs_corrector)
+            nmatches = len(ridx)
+
+        except Exception:
+            exc_type, exc_value, exc_tb = sys.exc_info()
+            traceback.print_exception(exc_type, exc_value, exc_tb, file=sys.stdout)
+            sys.exit()
+
+        assert (nmatches > 0.8*num_expected)
+
+
+    @pytest.mark.parametrize("input_filenames",
+                                [('j8ep04lwq')]
+                            )
+    def test_pipeline(self, input_filenames):
+        """Test of new pipeline alignment components (call separately)
+
+        This test performs separate fits to each chip separately.
+
+        Success Criteria
+        -----------------
+          * nmatches > 0 on all chips
+          * xrms and yrms < 30mas on all chips
+
+        """
+        self.input_loc = 'catalog_tests'
+        self.curdir = os.getcwd()
+        truth_path = [self.input_repo, self.tree, self.input_loc, *self.ref_loc]
+
+        if not isinstance(input_filenames, list):
+            input_filenames = [input_filenames]
+
+        # Use this to collect all failures BEFORE throwing AssertionError
+        failures=False
+        try:
+            # Make local copies of input files
+            local_files = []
+            for infile in input_filenames:
+                downloaded_files = self.get_input_file(infile, docopy=True)
+                local_files.extend(downloaded_files)
+
+            # generate reference catalog
+            refcat = amutils.create_astrometric_catalog(local_files, catalog='GAIADR2')
+
+            # Generate source catalogs for each input image
+            source_catalogs = generate_source_catalogs(local_files)
+
+            # Convert input images to tweakwcs-compatible NDData objects and
+            # attach source catalogs to them.
+            imglist = []
+            for group_id,image in enumerate(local_files):
+                imglist.extend(amutils.build_nddata(image, group_id,
+                                                    source_catalogs[image]['catalog_table']))
+
+            # Specify matching algorithm to use
+            match = tweakwcs.TPMatch(searchrad=250, separation=0.1,
+                                     tolerance=5, use2dhist=True)
+
+            # Align images and correct WCS
+            tweakwcs.tweak_image_wcs(imglist, refcat, match=match)
+            for chip in imglist:
+                status = chip.meta['tweakwcs_info']['status']
+                if status.startswith('FAIL'):
+                    failures = True
+                    print("STATUS for {}: {}".format(chip.meta['filename'], status))
+                    break
+        except Exception:

```

Comment:
Blank exceptions are bad. Any failue real or imaged will simply write out `Failed to align xyz.ext`. At the very least do this:

```python
try:
    # something
except Exception as e:
    print("ALIGNMENT EXCEPTION: Failed to align {} due to {}".format(infile, e))
```

Reply:
Done

Reply:
This entire Python file is deprecated and will be deleted.